### PR TITLE
Stop logging the no space left on device error

### LINKED ISSUE #208: Stop logging all the errors that come from LMDB
To avoid sending too many errors to `sentry` we would like to stop sending errors related to LMDB.

See : _@qdequele comments in https://github.com/meilisearch/transplant/issues/180#issuecomment-860690861_

### PR DESCRIPTION
**Pull request by irevoire**
closes #208
@qdequele what do you think of that?
Are there any other errors we need to ignore?

As you can see in the code, once we are in `Sentry` the error has already been converted to a `String` so the only thing we can do to see if we need to send the error or not is to match the `String` against our error message. 
If we have a lot of other logs we want to ignore I would suggest prefixing all the logs with something like:
```
User error: No space left on device
```
So in Sentry, we could just check if the log start by `User error:` and ignore all these errors at once

### CODE REVIEW

File: meilisearch-http/src/main.rs [hunk 1]
```diff
@@ -46,6 +46,14 @@ async fn main() -> Result<(), MainError> {
                 let sentry = sentry::init(sentry::ClientOptions {
                     release: sentry::release_name!(),
                     dsn: Some(SENTRY_DSN.parse()?),
+                    before_send: Some(Arc::new(|event| {
+                        event
+                            .message
+                            .as_ref()
+                            .map(|msg| msg.contains("no space left on device"))
```
Comment (Kerollmops) [inline meilisearch-http/src/main.rs:53]:
```suggestion
                            .map(|msg| msg.to_lowercase().contains("no space left on device"))
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]

**Comment by qdequele** [comment]
It is not possible to not log the "no space left" errors as real errors and log them as traces logs?

**Comment by irevoire** [comment]
Yes, we could do that, but as a user, that's probably one of the few logs that would really interest me since it stops me from using meilisearch.
In my opinion, this log should say a big red ERROR.

But if it was the only error, you wanted to remove, we can merge this PR as-is and it'll be ok :smile:

**Comment by curquiza** [comment]
I agree with @irevoire. Plus, the error of "no space left on the device" on windows should not be present anymore 🤞
If it's still really annoying @qdequele, let us know once the v0.21 is out!

**Review by curquiza — APPROVED** [review]
thanks @irevoire!

bors merge

**Pull request merged**